crates/core/src/client/mod.rs

```diff
@@ -97,12 +97,10 @@ impl<T: JsonRpcTransport + Send + Sync> KakarotEthApi<T> for KakarotClient<JsonR
         let starknet_contract_address = self.starknet_provider.call(request, starknet_block_id).await?;
 
         // shadow the variable to FielElement from a Vec<FieldElement>, for use in subsequent code
-        let starknet_contract_address = match starknet_contract_address.get(0) {
+        let starknet_contract_address = match starknet_contract_address.first() {
             Some(x) if starknet_contract_address.len() == 1 => *x,
             _ => {
-                return Err(EthApiError::<T::Error>::OtherError(anyhow::anyhow!(
-                    "Kakarot get_code: starknet_contract_address is empty"
-                )));
+                return Err(DataDecodingError::InvalidReturnArrayLength("starknet_contract_address".into()).into());
```

Comment:
should we say:
`DataDecodingError::InvalidReturnArrayLength("starknet_contract_address")` -> `DataDecodingError::InvalidReturnArrayLength("compute_starknet_address should return an field element array of length 1, received starknet_contract_address length: {starknet_contract_address.len()}")`

Reply:
we can, I find it hard to balance how much information is needed in an error message, I started with somethink like
```
DataDecodingError::InvalidReturnArrayLength{expected: usize, actual: usize}
```